### ISSUE TITLE
Any way to control Docker version?

### ISSUE THREAD

**Issue by codenrhoden** [vmware/simple-k8s-test-env#6]
After using sk8 to successfully deploy a K8s cluster on VMC (PhotonOS), the resulting version of Docker shows as `17.06-ce`.

It was brought up to me today that recommended version of docker is [18.09](https://kubernetes.io/docs/setup/cri/#docker). Is there a way to have sk8 install a newer version? or is this a limitation of what is packaged for Photon?

I have no indication at this point that anything is *wrong* with `17.06-ce`. Just something I wanted to bring up.

**Comment by akutz** [comment]
Hi @codenrhoden,

sk8 does not use the Docker installation that ships with Photon. Instead sk8 deploys containerd as the CRI.

**Comment by akutz** [comment]
Hi @codenrhoden,

Here's a figure depicting the components installed by sk8:

![Node components](https://s3-us-west-2.amazonaws.com/cnx.vmware/cicd/sk8/svg/node-components.svg?v2)

**Comment by codenrhoden** [comment]
Yeah, I looked at these before I asked. I was wondering about just that very thing. I wonder how docker gets installed? It must get pulled in by a dependency...

```
$ docker version
Client:
 Version:      17.06.0-ce
 API version:  1.30
 Go version:   go1.9.4
 Git commit:   02c1d87
 Built:        Fri Sep 21 13:28:59 2018
 OS/Arch:      linux/amd64
Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
```

So we can see that `dockerd` isn't running. So that version is just the docker CLI. Oddly the daemon is still installed, even though `runc` is what  actually launches the images these days, even under `dockerd`.  So it's still coming in somehow. And I wonder if there is a way to up the version of `runc` or something?

```
$ ls -al /usr/bin/dock*
-rwxr-xr-x 1 root root 19257240 Sep 21 13:29 /usr/bin/docker
-rwxr-xr-x 1 root root 11897720 Sep 21 13:30 /usr/bin/docker-containerd
-rwxr-xr-x 1 root root 10952632 Sep 21 13:30 /usr/bin/docker-containerd-ctr
-rwxr-xr-x 1 root root  2795080 Sep 21 13:30 /usr/bin/docker-containerd-shim
-rwxr-xr-x 1 root root 38978688 Sep 21 13:31 /usr/bin/dockerd
-rwxr-xr-x 1 root root   667552 Sep 21 13:30 /usr/bin/docker-init
-rwxr-xr-x 1 root root  2539848 Sep 21 13:30 /usr/bin/docker-proxy
```

Like I said, I'm not saying anything is wrong at all. Just trying to tease out what's happening under the covers. =)
